lib/utils.js

```diff
@@ -183,24 +183,9 @@ utils.getMapValueByKey = (obj, key) => {
   return getMapValue(obj, key);
 };
 
-utils.addExtensions = (obj) => {
-  obj.prototype.extensions = function () {
-    const result = {};
-    Object.entries(this._json).forEach(([key, value]) => {
-      if ((/^x-[\w\d\.\-\_]+$/).test(key)) {
-        result[String(key)] = value;
-      }
-    });
-    return result;
-  };
-  
-  obj.prototype.ext = function (name) {
-    return this._json[String(name)];
-  };
-  
-  obj.prototype.extension = obj.prototype.ext;
-
-  return obj;
+utils.mix = (model, ...mixins) => {
+  mixins.forEach(mixin => Object.assign(model.prototype, mixin));
```

Comment:
Mixins should be only applied for models and I think that the specification itself shouldn't use duplicate names for different values, e.g: `externalDocs` in 99% models for reference of `ExternalDocs` object and in one model for inline string value with url. Stupid example, but I think that you understand me.

Reply:
not really 😄 

I mean, for example, how we can validate that by mistake we will not add again for example a `description` function to Info model

Reply:
Then we should write our `Object.assign` or another extend function and check the duplication :)

Reply:
please update jsdoc so I know what you pass to the function

Reply:
looking at the input arguments, it would be possible to validate if function from `mixins` is already present in the `model` right? and throw error in case it is, then our tests would always fail if someone by mistake makes a duplication. Or you think it is super bad idea to secure it this way?

Reply:
Added validation, but I hardcoded error message inside function. Is it ok, or I must export error's message to separate file/const?

Reply:
I'm more concerned about the message itself, it is not describing well what happens, that there is a function duplication happening. 

Just to make sure, now, if you would add for example `description` function to some model that already uses mixin with this function, tests will fail?

Reply:
I added appropriate test and change logic. Already function first check duplication of model in mixins array, then check if model has at least one method (name) from given mixins.